ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/activeMapping/ContinuousPlannerSchedulingTask.java

```diff
@@ -46,321 +31,112 @@
+      State readyToPlanState = new ReadyToPlanState(commandMessage,
+                                                    continuousPlanner,
+                                                    controllerFootstepQueueMonitor,
+                                                    continuousHikingParameters,
+                                                    terrainMap,
+                                                    debugger,
+                                                    statistics,
+                                                    stepValidityChecker);
+      State waitingtoLandState = new WaitingToLandState(ros2Helper,
+                                                        simpleRobotName,
+                                                        continuousPlanner,
+                                                        controllerFootstepQueueMonitor,
+                                                        continuousHikingParameters,
+                                                        statistics);
+
+      // Adding the different states
+      stateMachineFactory.addState(ContinuousHikingState.DO_NOTHING, notStartedState);
+      stateMachineFactory.addState(ContinuousHikingState.WAITING_TO_LAND, waitingtoLandState);
+      stateMachineFactory.addState(ContinuousHikingState.READY_TO_PLAN, readyToPlanState);
+
+      // Create different conditions
+      StartContinuousHikingTransitionCondition startContinuousHikingTransitionCondition = new StartContinuousHikingTransitionCondition(commandMessage,
+                                                                                                                                       continuousHikingParameters);
+      StopContinuousHikingTransitionCondition stopContinuousHikingTransitionCondition = new StopContinuousHikingTransitionCondition(continuousHikingParameters,
+                                                                                                                                    commandMessage);
+      PlanAgainTransitionCondition planAgainTransitionCondition = new PlanAgainTransitionCondition(continuousPlanner, continuousHikingParameters);
+
+      // From any given state we can go back to DO_NOTHING and stop ContinuousHiking
+      stateMachineFactory.addTransition(ContinuousHikingState.WAITING_TO_LAND, ContinuousHikingState.DO_NOTHING, stopContinuousHikingTransitionCondition);
+      stateMachineFactory.addTransition(ContinuousHikingState.READY_TO_PLAN, ContinuousHikingState.DO_NOTHING, stopContinuousHikingTransitionCondition);
+
+      // Add condition, this triggers the state machine to start Continuous Hiking
+      stateMachineFactory.addTransition(ContinuousHikingState.DO_NOTHING, ContinuousHikingState.READY_TO_PLAN, startContinuousHikingTransitionCondition);
+
+      // Add condition, this allows us to plan over and over again without sending any footsteps to the controller
```

Comment:
I added the comments because I found them helpful, especially if I have to look back at it. Once you get your head around the IHMC State Machine its not too bad, however for newer people its a little complex because some things happen behind the scenes